BetaCameras/Hikvision/H264Decoder.cs

```diff
@@ -0,0 +1,213 @@
+            if (initialized)
+            {
+                //Sequence of freeing all ffmpeg resources and calling constructor again caused an error in avcodec_send_packet in rare cases.
+                //Thus, the init sequence can only be called once.
+                //Important: Avoiding the re-init only works if the firstCompressedPacket contains a key frame.
+                //This should be guaranteed, since the H264 encoding on server side is re-initialized, too.
+                return;
+            }
+
+            initialized = true;
+
+            ffmpeg.av_register_all();
+
+            int dataSize = 0;
+            foreach (byte[] nalUnit in nalUnits)
+            {
+                dataSize += nalUnit.Length + 4;
+            }
+
+            byte* dat = (byte*)ffmpeg.av_malloc((ulong)dataSize);
+
+
+            fixed (byte* start = startSequence)
+            {
+                foreach (byte[] nalUnit in nalUnits)
+                {
+                    fixed (byte* dataPtr = nalUnit)
+                    {
+                        UnmanagedMemory.CopyMemory(dat, start, (uint)startSequence.Length);
+                        dat += startSequence.Length;
+                        UnmanagedMemory.CopyMemory(dat, dataPtr, (uint)nalUnit.Length);
+                        dat += nalUnit.Length;
+                    }
+                }
+
+                dat -= dataSize;
+            }
+
+            AVFormatContext* icLocal = ffmpeg.avformat_alloc_context();
+
+            ic = icLocal;
+
+            avio_alloc_context_write_packet_func writeCallback;
+            writeCallback.Pointer = IntPtr.Zero;
+            avio_alloc_context_seek_func seekCallback;
+            seekCallback.Pointer = IntPtr.Zero;
+            avio_alloc_context_read_packet_func readCallback;
+            readCallback.Pointer = IntPtr.Zero;
+
+            icLocal->pb = ffmpeg.avio_alloc_context(dat, bufferSize, 0, null, readCallback, writeCallback, seekCallback);
+
+            if (icLocal->pb == null)
+            {
+                throw new Exception("Failed to allocate ffmpeg context.");
+            }
+
+            // Need to probe buffer for input format unless you already know it
+            AVProbeData probe_data;
+            probe_data.buf_size = dataSize;
+            probe_data.filename = (byte*)Marshal.StringToHGlobalAnsi("stream");
+            probe_data.buf = (byte*)UnmanagedMemory.Alloc(probe_data.buf_size);
+            UnmanagedMemory.CopyMemory(probe_data.buf, dat, (uint)probe_data.buf_size);
+
+            AVInputFormat* pAVInputFormat = ffmpeg.av_probe_input_format(&probe_data, 1);
+
+            if (pAVInputFormat == null)
+            {
+                pAVInputFormat = ffmpeg.av_probe_input_format(&probe_data, 0);
+            }
+
+            // cleanup
+            UnmanagedMemory.DeAlloc((IntPtr)probe_data.buf, probe_data.buf_size);
+            probe_data.buf = null;
+
+            pAVInputFormat->flags |= ffmpeg.AVFMT_NOFILE;
```

Comment:
I'm always getting a NRE here (because `pAVInputFormat` is null), even if the application directory contains all the native FFMPEG dependencies. Do you have an idea, why that happens?

Reply:
Problem was that the camera was configured to H265.